policy/watcher.go

```diff
@@ -486,31 +523,89 @@ func (w *watcher) defaultPolicyAction(action string) (map[string]Rule, error) {
 	return defPolicyActionMap, nil
 }
 
+func (w *watcher) defaultPolicyActionOut(action string) (map[string]Rule, error) {
+	defPolicyActionMap := make(map[string]Rule)
+
+	ruleName := "all.local.containers.out"
+	all := w.getAllLocalContainers()
+	isStateful := false
+	r, err := w.buildAndProcessRuleWithSrcDstExt(isStateful, ruleName, nil, all)
+	if err != nil {
+		return nil, err
+	}
+	r.action = action
+	defPolicyActionMap[ruleName] = *r
+
+	logrus.Debugf("defPolicyActionMap: %v", defPolicyActionMap)
+	return defPolicyActionMap, nil
+}
+
 func (w *watcher) defaultSystemStackPolicies() (map[string]Rule, error) {
 	defSysRulesMap := make(map[string]Rule)
 	for _, stack := range w.stacks {
 		if !stack.System {
 			continue
 		}
 
-		_, all := w.getInfoFromStack(stack)
-
-		ruleName := fmt.Sprintf("from.system.stack.%v", stack.Name)
-		isStateful := false
-		isDstSystem := true
-		isSrcSystem := true
-		r, err := w.buildAndProcessRuleWithSrcDst(isStateful, isDstSystem, isSrcSystem, ruleName, nil, all)
-		if err != nil {
-			return nil, err
+		local, all := w.getInfoFromStack(stack)
+		if len(all) > 0 {
+			ruleName := fmt.Sprintf("from.system.stack.%v", stack.Name)
+			isStateful := false
+			isDstSystem := true
+			isSrcSystem := true
+			r, err := w.buildAndProcessRuleWithSrcDst(isStateful, isDstSystem, isSrcSystem, ruleName, nil, all)
+			if err != nil {
+				return nil, err
+			}
+			r.action = ActionAllow
+			defSysRulesMap[ruleName] = *r
+		}
+		if len(local) > 0 {
+			ruleName := fmt.Sprintf("ex.fr.system.stack.%v", stack.Name)
+			isStateful := false
```

Comment:
If we decide to allow only outgoing connections to external IP addresses, this would become stateful.